src/normal.c

```diff
@@ -5782,6 +5782,10 @@ nv_g_home_m_cmd(cmdarg_T *cap)
 	curwin->w_valid &= ~VALID_WCOL;
     }
     curwin->w_set_curswant = TRUE;
+#ifdef FEAT_FOLDING
+    if (curwin->w_cline_folded)
```

Comment:
`validate_cheight()` may need to be called before using this value.

Reply:
yeah right

Reply:
done